### DISCUSSION TITLE
How can I export GraphQL API query results in i.e. Excel as a CSV?

### DISCUSSION THREAD

**Discussion by Ranicki69** [community/community#24344] [Apps, API and Webhooks]
<p>Is there an easy way to export the data received so it can be used in Excel?</p>
<p>Thank you in advance!</p>

**Comment by lee-dohm** [comment]
<p>Unfortunately, not really. CSV is designed for tabular data, in other words it is designed for data formatted as columns and rows. JSON, the kind of data that GraphQL returns, is designed for hierarchical data, in other words it is designed for data formatted as a set of hierarchical, nested objects.</p>
<p>If you formatted your query very specifically, to only return flat rows, or if you post-processed the information you received to turn it from hierarchical data to tabular data, then you could convert it to CSV. But there’s no simple way to take the result of any GraphQL query and convert it to CSV.</p>
<p>I hope that helps. Let us know if you have more questions.</p>

**Comment by commonsensej** [comment]
<p>Awesome online tool to convert GraphQL JSON to CSV:<br>
<a href="https://konklone.io/json/" rel="nofollow noopener">https://konklone.io/json/</a></p>

**Reply by MoysesPimenta** [reply]
duude you just saved me hoursss of my life thanks a lot

**Reply by boudra** [reply]
if someone is working with sensitive data or feel uneasy using a browser based converter, you can use Konbert desktop: https://konbert.com/desktop

**Comment by nethgato** [comment, marked as answer]
<p>Just following up on this thread to reiterate that <a class="mention" href="/lee-dohm">@lee-dohm</a>’s post is accurate to what’s available within the GitHub platform itself.</p>
<p>Though I do love <a class="mention" href="/commonsensej">@commonsensej</a>’s suggested conversion tool! However, there is some concern there about potentially converting sensitive data. Which furthers the need to an official GitHub offering for JSON &gt;&gt; CSV formatting.</p>
<p>If anyone has this need and would like to see this built into GitHub itself, I strongly recommend submitting your suggestion via our feedback form, here:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://support-assets.github.com/assets/favicon-3dcd2c977a91b2e3f396b300c5ff0d572295ca632ddb7ba1adcacc1ea68dd5c0.ico" class="site-icon" width="16" height="16">
      <a href="https://support.github.com/contact/feedback" target="_blank" rel="noopener">support.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="16" height="16">

<h3><a href="https://support.github.com/contact/feedback" target="_blank" rel="noopener">Share feedback - GitHub Support</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>:bow:</p>

**Comment by joshjohanning** [comment]
If you're using the `gh api graphql` command, you can convert to a CSV format relatively easily with the `--jq` arg.

```sh
gh api graphql --paginate -F owner="joshjohanning-org" -f query='
query ($owner: String!, $endCursor: String) {
  organization(login: $owner) {
    repositories(first: 100, after: $endCursor) {
      nodes {
        name
        pushedAt
      }
      pageInfo {
        endCursor
        hasNextPage
      }
    }
  }
}' --jq '.data.organization.repositories.nodes[] | [.name, .pushedAt] | @tsv' > file.tsv
```

You can also use `@csv`:

```sh
gh api graphql --paginate -F owner="joshjohanning-org" -f query='
query ($owner: String!, $endCursor: String) {
  organization(login: $owner) {
    repositories(first: 100, after: $endCursor) {
      nodes {
        name
        pushedAt
      }
      pageInfo {
        endCursor
        hasNextPage
      }
    }
  }
}' --jq '.data.organization.repositories.nodes[] | [.name, .pushedAt] | @csv' > file.csv
```
